### PR TITLE
Use OTP's get_doc if available

### PR DESCRIPTION
**Pull request by dgud**
EEP48 provides documentation for erlang application/modules.
OTP-23 implements EEP48 and provide the doc info if available.

If user uses an OTP version that provides the functionality and the
doc_chunks are available use that by default and if not fallback to
fetching the docs from src via 'erlang_ls' own implementation.

This is a foundation for further work where we should write an
doc_chunk to markdown renderer, when markdown available in the client.

### CODE REVIEW

**Comment by dgud** [comment]
Hmm I don't understand while the tests are failing

**Comment by robertoaloi** [comment]
Having a look to the failing test cases...

**Comment by robertoaloi** [comment]
@dgud It looks like the property-based tests are complaining about the `halt` function not being called correctly. I tried running them locally for your branch and they pass. I suspect there's some flakyness around the `shutdown` solution. @jfacorro Do you have any more input?

https://github.com/erlang-ls/erlang_ls/pull/536/checks?check_run_id=494725902

**Comment by dgud** [comment]
Yeah I ran the tests locally before pushing the PR and that worked for me as well.

**Comment by jfacorro** [comment]
I've seen the GitHub job fail from time to time for no obvious reason. My approach so far has been to re-trigger the job, but I just did that and it seems the action is not able to checkout the reference because it's not in the tree:
```
From https://github.com/erlang-ls/erlang_ls
 * [new branch]      285-indexing-progress-reports -> origin/285-indexing-progress-reports
 * [new branch]      285-show-progress-report-during-indexing -> origin/285-show-progress-report-during-indexing
 * [new branch]      458-auto-complete-exported-types -> origin/458-auto-complete-exported-types
 * [new branch]      478-elvis-diagnostics-fail -> origin/478-elvis-diagnostics-fail
 * [new branch]      features-list          -> origin/features-list
 * [new branch]      indexing-no-compressed -> origin/indexing-no-compressed
 * [new branch]      intellij               -> origin/intellij
 * [new branch]      master                 -> origin/master
 * [new branch]      notes                  -> origin/notes
 * [new tag]         0.1.0                  -> 0.1.0
git checkout --progress --force 200ce9fbf446a9f557fe655ab875461620d16fef
##[error]fatal: reference is not a tree: 200ce9fbf446a9f557fe655ab875461620d16fef
Removed matchers: 'checkout-git'
##[error]Git checkout failed with exit code: 128
##[error]Exit code 1 returned from process: file name '/home/runner/runners/2.165.2/bin/Runner.PluginHost', arguments 'action "GitHub.Runner.Plugins.Repository.v1_0.CheckoutTask, Runner.Plugins"'.
```
I'm not sure what's happening there 😞.

**Comment by jfacorro** [comment]
@dgud Could you rebase on `master`? I made a change to the build [here](https://github.com/erlang-ls/erlang_ls/pull/543) so that it at least doesn't fail because of the `Checkout` action.

File: src/els_hover_provider.erl [hunk 1]
```diff
@@ -57,6 +57,33 @@ documentation(_M, _POI) ->
...
+            docs_from_src(M, F, A);
+          FuncDoc ->
+            #{ kind => content_kind()
+             , value => unicode:characters_to_binary(FuncDoc)
+             }
+        end;
+    _R1 ->
+      docs_from_src(M, F, A)
+  catch error:undef ->
+      lager:info("err: ~p~n", [undef]),
```
Comment (robertoaloi) [inline src/els_hover_provider.erl]: We should probably use a `warning` level here and a nicer error message. Also, no need for newline.

```suggestion
      lager:warning("Error fetching documentation. Using an old OTP version?", []),
```

File: src/els_hover_provider.erl [hunk 2]
```diff
@@ -57,6 +57,33 @@ documentation(_M, _POI) ->
 %% @doc get the docs
 -spec get_docs(atom(), atom(), byte()) -> binary().
 get_docs(M, F, A) ->
+  %% Use apply and hard coded record so it can be compiled on older OTP's
```
Comment (robertoaloi) [inline src/els_hover_provider.erl]: We should probably be more specific on what "older" means here. I suggest we update the `edoc` for this function with a nicer description, mentioning the minimum required OTP version for this to work.

File: src/els_hover_provider.erl (hunk 1 above)
Comment (dgud) [reply src/els_hover_provider.erl]: That is a debug printout used during development, so IMHO it should be removed.

Comment (robertoaloi) [reply src/els_hover_provider.erl]: Fine with that, too.

**Comment by robertoaloi** [comment]
Thanks @dgud ! 🚀

**Pull request merged**